addons/mail/static/tests/new/discuss/discuss_tests.js

```diff
@@ -884,3 +884,201 @@ QUnit.test("basic top bar rendering", async function (assert) {
+        message.querySelectorAll(".o-mail-message-actions .cursor-pointer").length,
+        4
+    );
+    assert.strictEqual(message.querySelectorAll(".o-mail-message-action-add-reaction").length, 1);
+    assert.strictEqual(message.querySelectorAll(".o-mail-message-action-toggle-star").length, 1);
+    assert.strictEqual(message.querySelectorAll(".o-mail-message-action-reply-to").length, 1);
+    assert.strictEqual(message.querySelectorAll(".o-mail-message-action-mark-read").length, 1);
+});
+
+QUnit.test('messages marked as read move to "History" mailbox', async function (assert) {
+    const pyEnv = await startServer();
+    const mailChannelId1 = pyEnv["mail.channel"].create({ name: "other-disco" });
+    const [mailMessageId1, mailMessageId2] = pyEnv["mail.message"].create([
+        {
+            body: "not empty",
+            model: "mail.channel", // value to link message to channel
+            needaction: true,
+            res_id: mailChannelId1, // id of related channel
+        },
+        {
+            body: "not empty",
+            model: "mail.channel", // value to link message to channel
+            needaction: true,
+            res_id: mailChannelId1, // id of related channel
+        },
+    ]);
+    pyEnv["mail.notification"].create([
+        {
+            mail_message_id: mailMessageId1, // id of related message
+            notification_type: "inbox",
+            res_partner_id: pyEnv.currentPartnerId, // must be for current partner
+        },
+        {
+            mail_message_id: mailMessageId2, // id of related message
+            notification_type: "inbox",
+            res_partner_id: pyEnv.currentPartnerId, // must be for current partner
+        },
+    ]);
+    const { click, openDiscuss } = await start({
+        discuss: {
+            params: {
+                default_active_id: "mail.box_history",
+            },
+        },
+    });
+    await openDiscuss();
+    assert.ok(
+        target.querySelector('button[data-mailbox="history"]').classList.contains("o-active")
+    );
+    assert.strictEqual(
+        target.querySelectorAll('.o-mail-discuss-content .o-mail-thread [data-empty-thread=""]')
+            .length,
+        1
+    );
+
+    await click('button[data-mailbox="inbox"]');
+    assert.ok(target.querySelector('button[data-mailbox="inbox"]').classList.contains("o-active"));
+    assert.strictEqual(
+        target.querySelectorAll('.o-mail-discuss-content .o-mail-thread [data-empty-thread=""]')
+            .length,
+        0
+    );
+    assert.strictEqual(
+        target.querySelectorAll(".o-mail-discuss-content .o-mail-thread .o-mail-message").length,
+        2
+    );
+
+    await click('.o-mail-discuss-actions button[data-action="mark-all-read"]');
+    assert.ok(target.querySelector('button[data-mailbox="inbox"]').classList.contains("o-active"));
+    assert.strictEqual(
+        target.querySelectorAll('.o-mail-discuss-content .o-mail-thread [data-empty-thread=""]')
```

Comment:
I slightly prefer `$(target).find()` over `target.querySelector()`, because the former has less token for the technical stuffs, thus the meaningful part is more apparent.